File: seed/rust-sdk/trace/src/types/test_submission_status.rs
use crate::error_info::ErrorInfo;
use crate::running_submission_state::RunningSubmissionState;
use std::collections::HashMap;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(tag = "type")]
pub enum TestSubmissionStatus {
        Stopped,

        Errored {
            value: ErrorInfo,
        },

        Running {
            value: RunningSubmissionState,
        },

        TestCaseIdToState {
            value: HashMap<String, SubmissionStatusForTestCase>,
        },
}

File: seed/ruby-sdk-v2/trace/lib/fern_trace/submission/types/test_submission_status.rb
# frozen_string_literal: true

module FernTrace
  module Submission
    module Types
      class TestSubmissionStatus < Internal::Types::Model
        extend FernTrace::Internal::Types::Union

        discriminant :type

        member -> { Object }, key: "STOPPED"
        member -> { FernTrace::Submission::Types::ErrorInfo }, key: "ERRORED"
        member -> { FernTrace::Submission::Types::RunningSubmissionState }, key: "RUNNING"
        member -> { Internal::Types::Hash[String, FernTrace::Submission::Types::SubmissionStatusForTestCase] }, key: "TEST_CASE_ID_TO_STATE"
      end
    end
  end
end
